system/lib/fetch/wasmfs.cpp

```diff
@@ -0,0 +1,55 @@
+// // Copyright 2021 The Emscripten Authors.  All rights reserved.
+// // Emscripten is available under two separate licenses, the MIT license and the
+// // University of Illinois/NCSA Open Source License.  Both these licenses can be
+// // found in the LICENSE file.
+
+#include <stdlib.h>
+#define __NEED_struct_iovec
+#include <emscripten/emscripten.h>
+#include <wasi/api.h>
+
+extern "C" {
+
+  __wasi_errno_t __wasi_fd_write( __wasi_fd_t fd, const __wasi_ciovec_t *iovs, size_t iovs_len, __wasi_size_t *nwritten) {
+    if (fd == 1 || fd == 2) {
+      MAIN_THREAD_EM_ASM({
+        var buffer = [];
+        var fd = $0;
+        var iovs = $1;
+        var iovs_len = $2;
+        var nwritten = $3;
+        var num = 0;
+        for (var i = 0; i < iovs_len; i++) {
+          var ptr = HEAP32[(((iovs)+(i*8))>>2)];
+          var len = HEAP32[(((iovs)+(i*8 + 4))>>2)];
+          for (var j = 0; j < len; j++) {
+            if (HEAPU8[ptr+j] === 0 || HEAPU8[ptr+j] === 10) {
+              (fd === 1 ? out : err)(UTF8ArrayToString(buffer, 0));
+              buffer.length = 0;
+            } else {
+              buffer.push(HEAPU8[ptr+j]);
+            }
+          }
+          num += len;
+        }
+        HEAP32[((nwritten)>>2)] = num;
+      }, fd, iovs, iovs_len, nwritten);
+    }
+    return 0;
+  }
+
+  __wasi_errno_t __wasi_fd_seek(__wasi_fd_t fd, __wasi_filedelta_t offset, __wasi_whence_t whence, __wasi_filesize_t *newoffset)
+  {
+    abort();
```

Comment:
Might be nice to make a function that gets a name and prints an error out before aborting, so the runtime error is a little clearer. But it would help mostly you @ethanalee so up to you...

Reply:
I can have it print out the name of the stubbed function? @kripken

Reply:
I think you can: https://gcc.gnu.org/onlinedocs/gcc/Function-Names.html

Reply:
awesome!